src/engine/test/helper_tests/run.py

```diff
@@ -112,73 +124,358 @@ def load_yaml(file_path):
             return yaml.safe_load(stream)
         except yaml.YAMLError as exc:
             print(exc)
+            sys.exit(1)
 
 
 def string_to_resource_type(string_value):
+    """
+    Converts a string value to a corresponding ResourceType enumeration value.
+
+    Args:
+        string_value (str): The string representation of the resource type.
+
+    Returns:
+        api_catalog.ResourceType: The corresponding ResourceType value.
+        If the string value is invalid, returns api_catalog.ResourceType.UNKNOWN.
+    """
     try:
         return api_catalog.ResourceType.Value(string_value)
     except ValueError:
         return api_catalog.ResourceType.UNKNOWN
 
 
 def string_to_resource_format(string_value):
+    """
+    Converts a string value to a corresponding ResourceFormat enumeration value.
+
+    Args:
+        string_value (str): The string representation of the resource format.
+
+    Returns:
+        api_catalog.ResourceFormat: The corresponding ResourceFormat value.
+        If the string value is invalid, returns api_catalog.ResourceFormat.json.
+    """
     try:
         return api_catalog.ResourceFormat.Value(string_value)
     except ValueError:
         return api_catalog.ResourceFormat.json
 
```

Comment:
This only call once with json as str argument, use the ResourceFormatEnum instead.